api/models/prometheus.go

```diff
@@ -0,0 +1,101 @@
+package models
+
+import (
+	"sort"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/storage"
+)
+
+type PrometheusQueryRange struct {
+	Query   string `form:"query"`   //<string>: Prometheus expression query string.
+	Start   string `form:"start"`   //<rfc3339 | unix_timestamp>: Start timestamp.
+	End     string `form:"end"`     //<rfc3339 | unix_timestamp>: End timestamp.
+	Step    string `form:"step"`    //<duration>: Query resolution step width.
+	Timeout string `form:"timeout"` //<duration>: Evaluation timeout. Optional. Defaults to and is capped by the value of the -query.timeout flag.
```

Comment:
what does `duration` mean here? is it an integer or string format like `10s` ? is the format documented somewhere?

Reply:
It's an integer in seconds

Reply:
then why is the type string?

Reply:
It can be given in the form `10s` or `1m` so it needs to be in string format